monkey/monkey_island/cc/services/attack/technique_reports/T1504.py

```diff
@@ -25,14 +24,23 @@ class T1504(AttackTechnique):
     def get_report_data():
         data = {'title': T1504.technique_title(), 'info': []}
 
-        powershell_profile_modification_info = list(mongo.db.telemetry.aggregate(T1504.query))
+        shell_startup_files_modification_info = list(mongo.db.telemetry.aggregate(T1504.query))
+
+        powershell_startup_modification_info = []
+        for shell_startup_file_result in shell_startup_files_modification_info[0]['result']:
+            # only want powershell startup files
+            if "profile.ps1" in shell_startup_file_result[0]:
```

Comment:
Search for "profile.ps1" via mongodb query?

Reply:
In response to this and the comment below -

I'm not sure if I can do that because the telemetry data looks something like this (only kept the relevant fields so it's easier to read):
```python3
 [{ 'data': { 'status': 2,
              'technique': 'T1064',
              'usage': 'Scripts were used to execute Execute command when a '
                       'particular signal is received post breach action.'},
    'telem_category': 'attack'},

  { 'data': { 'command': [ 'trap \'echo "Successfully used trap command"\' INT '
                           '&& kill -2 $$ ;',
                           'trap - INT'],
              'hostname': 'Shreya',
              'ip': '127.0.1.1',
              'name': 'Execute command when a particular signal is received',
              'result': ['Successfully used trap command\n', True]},
    'telem_category': 'post_breach'},

  { 'data': { 'command': '',
              'hostname': 'Shreya',
              'ip': '127.0.1.1',
              'name': 'Modify shell startup file',
              'result': [ [ '# Succesfully modified /home/shreya/.profile\n',
                            True],
                          [ '# Succesfully modified /home/shreya/.bashrc\n',
                            True]]},
    'telem_category': 'post_breach'}]
```
Is it possible for me to modify the query such that I can iterate inside `data.result` and pick out specific fields?

Reply:
https://github.com/guardicore/monkey/issues/637
It's also possible to `unwind` an array into separate documents that you can query.

Reply:
Done!